src/main/java/seedu/address/model/contact/ContactDisplaySetting.java

```diff
@@ -0,0 +1,113 @@
+package seedu.address.model.contact;
+
+import java.util.Objects;
+
+/**
+ * Contains all the display settings for contacts in the model manager.
+ */
+public class ContactDisplaySetting {
+    public static final ContactDisplaySetting DEFAULT_SETTING = new ContactDisplaySetting(false);
+    private final boolean willDisplayPhone;
+    private final boolean willDisplayEmail;
+    private final boolean willDisplayTelegramHandle;
+    private final boolean willDisplayAddress;
+    private final boolean willDisplayZoomLink;
+    private final boolean willDisplayTags;
+    private final boolean isViewingFull;
+
+    /**
+     * Creates a new {@code ContactDisplaySetting} with the given settings.
+     * This is usually used in {@code CListCommand}, where the {@code isViewingFull} is always false.
+     */
+    public ContactDisplaySetting(
+        boolean willDisplayPhone, boolean willDisplayEmail, boolean willDisplayTelegramHandle,
+        boolean willDisplayAddress,
+        boolean willDisplayZoomLink, boolean willDisplayTags) {
+        this.willDisplayPhone = willDisplayPhone;
+        this.willDisplayEmail = willDisplayEmail;
+        this.willDisplayTelegramHandle = willDisplayTelegramHandle;
+        this.willDisplayAddress = willDisplayAddress;
+        this.willDisplayZoomLink = willDisplayZoomLink;
+        this.willDisplayTags = willDisplayTags;
+        this.isViewingFull = false;
```

Comment:
would it be more logical to move this setting to be inside the individual contact instead of the whole contact list? but if this works for cview and eview, i guess its okay

Reply:
This is very similar to the old implementation, whereby the settings are stored as static variables. It is possible to have the individual contacts and events store these settings. But since we are trying to make contacts and events immutable, I think there will be a major performance cost involved, as each time we call those commands which will change the display settings, all events and contacts will need to be replaced.

Reply:
hmm if we only move the isViewing setting to be managed by individual contact/event, we would only need to re-create the specific contact/event for cview/eview right? but i also feel its a bit weird if only isViewing setting is separated from the rest of the setting. haha but yeah i think ur current implementation is fine too 👍